routes/api/stories.js

```diff
@@ -0,0 +1,74 @@
+//2
+const express = require('express');
+const router = express.Router();
+const createError = require('http-errors');
+const {asyncHandler} = require('../utils');
+const { PostLike } = require("../../db/models");
+const { restoreUser } = require('../../auth');
+
+/* Like a story */
+router.post('/stories/:id/likes', restoreUser, asyncHandler(async(req, res, next) => {
+
+        if (!res.locals.authenticated) {
+
+           /* If the user is not authenticated,
+              return a 401 error */ 
+           return next(createError(401));
+
+        } else if (res.locals.authenticated ) {
+            
+            /* If the user is authenticated,
+               create a new like */
+            const postLike = await PostLike.create({
+                userId: res.locals.user.id,
+                postId: req.params.id
+            })
+        }
+
+        /* Initialize new post like variable to pass as json response */
+        const newPostLike = await PostLike.findByPk(req.params.id);
+
+        /* User has been authenticated, and
+           a like has been created.
+           Set 201 status code for response, and
+           send newPostLike as json response. */
+        res.status(201).json(newPostLike);
```

Comment:
There is no need to query for the post that was created. If line 28 gets moved inside the scope of the `postLike` on like 22 that will work.

Reply:
The code has been refactored to remove the superfluous query and deliver the response within the scope of the created postLike.